kernel/nativelib.mli

```diff
@@ -40,7 +40,7 @@ val compile_library : native_library -> string -> unit
     in [rt1] and [rt2] *)
 val execute_library :
   prefix:string -> string -> Nativecode.code_location_updates ->
-  Nativevalues.t * Nativevalues.t
+  Nativevalues.t option * Nativevalues.t option
```

Comment:
Leaking this implementation detail does not feel quite right. You should move the calls to `Option.get` into `execute_library`.